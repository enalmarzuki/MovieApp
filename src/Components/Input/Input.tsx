import React from 'react';
import {TextInput} from 'react-native-paper';
import {colors} from '../../Utils/colors';

type TextInputProps = React.ComponentProps<typeof TextInput>;

type IInputProps = {
  label?: string;
  value: string;
  isRounded: boolean;
  keyboardType?:
    | 'default'
    | 'number-pad'
    | 'numeric'
    | 'email-address'
    | 'phone-pad';
  error?: boolean;
  onFocus?: (args: any) => void;
  onBlur?: (args: any) => void;
  onChangeText: (args: any) => void;
} & TextInputProps;

const Input: React.FC<IInputProps> = ({
  label,
  value,
  isRounded,
  keyboardType = 'default',
  error,
  onChangeText,
  onFocus,
  onBlur,
  ...props
}) => {
  return (
    <TextInput
      label={label || ''}
      value={value}
      keyboardType={keyboardType}
      onChangeText={onChangeText}
      onFocus={onFocus}
      onBlur={onBlur}
      error={error}
      underlineColor="transparent"
      activeUnderlineColor="transparent"
      placeholderTextColor="#BBBBBB"
      outlineColor="transparent"
      theme={{
        colors: {
          primary: '#fff',
          text: colors.White,
        },
        roundness: isRounded ? 30 : 10,
      }}
      {...props}
    />
  );
};

export default Input;
