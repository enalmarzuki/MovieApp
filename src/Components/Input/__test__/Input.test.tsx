import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import Input from '../Input';
jest.useFakeTimers();

jest.mock(
  '../../../../node_modules/react-native/Libraries/Animated/NativeAnimatedHelper',
);

describe('should render Input default', () => {
  it('should render correctly', () => {
    const mockFunction = jest.fn();
    const InputSnap = renderer
      .create(
        <Input
          isRounded
          mode="outlined"
          placeholder="Search..."
          value="mock value"
          onChangeText={mockFunction}
        />,
      )
      .toJSON();

    expect(InputSnap).toMatchSnapshot();
  });

  it('input must have placeholder Search...', () => {
    const mockFunction = jest.fn();
    const {getByPlaceholderText} = render(
      <Input
        isRounded
        mode="outlined"
        placeholder="Search..."
        value="mock value"
        onChangeText={mockFunction}
      />,
    );
    const placeholderText = getByPlaceholderText('Search...');
    expect(placeholderText).toBeTruthy();
  });

  it('should input have value Spiderman', () => {
    const mockFunction = jest.fn();
    const {getByTestId} = render(
      <Input
        isRounded
        mode="outlined"
        placeholder="Search..."
        value="mock value"
        onChangeText={mockFunction}
      />,
    );
    const inputComponent = getByTestId('text-input-outline');
    fireEvent.changeText(inputComponent, 'Spiderman');
    expect(mockFunction).toHaveBeenCalledWith('Spiderman');
  });
});
