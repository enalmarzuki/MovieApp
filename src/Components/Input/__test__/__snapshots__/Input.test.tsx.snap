// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Input default should render correctly 1`] = `
<View
  style={{}}
>
  <View
    pointerEvents="none"
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 6,
        },
        {
          "backgroundColor": "#f6f6f6",
          "borderColor": "transparent",
          "borderRadius": 30,
          "borderWidth": 1,
        },
      ]
    }
    testID="text-input-outline"
  />
  <View>
    <View
      style={
        [
          {
            "paddingBottom": 0,
          },
          {
            "minHeight": 64,
            "paddingTop": 8,
          },
        ]
      }
    >
      <TextInput
        activeUnderlineColor="transparent"
        editable={true}
        keyboardType="default"
        maxFontSizeMultiplier={1.5}
        multiline={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Search..."
        placeholderTextColor="#BBBBBB"
        selectionColor="#fff"
        style={
          [
            {
              "flexGrow": 1,
              "margin": 0,
              "paddingHorizontal": 14,
              "zIndex": 1,
            },
            {
              "height": 56,
            },
            {
              "paddingBottom": 0,
              "paddingTop": 0,
            },
            {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 16,
              "fontVariant": undefined,
              "fontWeight": undefined,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
            false,
            [
              {},
            ],
          ]
        }
        testID="text-input-outlined"
        underlineColorAndroid="transparent"
        value="mock value"
      />
    </View>
  </View>
</View>
`;
